### ISSUE TITLE
[question] Is the project still maintained ?

### ISSUE THREAD

**Issue by anthonytoktokdoc** [open-inc/node-parse-server-schema#2]
Hi,

First, thanks for the awesome work because it is really what I was looking for today ! cf. [post parse community](https://community.parseplatform.org/t/generate-ts-types-from-schema/2077)

I would like to know if this project is still maintained and if you accept contributions ? for ex: define for specific class/columns specific types (enums...). Also, can I link this package in my post ? I think it can be useful for others.

Thanks,

**Comment by zlls** [comment]
Hello Anthony,

thank you for your kind request. Yes this project is maintained. We are using this tool a lot in our internal projects. Feel free to link it in your post.

Be aware if using the up command, that this will delete columns if you change them in your local schema and will not alter them. This will cause loss of data. There is a flag to prevent this (in v2 there this flag will be default and you need a flag to "delete" data), but this is something I will aproach in the future by writing directly to the database.

The typescript command is currently write only. I would like to read the typescript code, if it exists, and add to it, so you can create an enum or specify what exaclty the array or object types may look like and run the typescript command again without losing any custom definitions. I didn't have time to do it yet.

Actually I would like to propose an idea to the parse server team, where we can add custom meta data to the schema, which can be pulled together with the schema definition. This way one could automate more sophisticated validations, code generation or generate a custom ui from just the schema (a bit like parse-dashboard, but more end user friendly.

With kind regards
Sebastian

**Comment by anthonytoktokdoc** [comment]
Hi Sebastian,

Thanks for you answers and ideas

I was thinking of making a PR with some of these options available to the typescript generation that we are using in our team. I don't know if it is really useful as it is somewhat related to our existing JS codebase.

**extendCollectionFrom**: mapping which specify if your class extend from another class like {className: string, from: string, path: string }

**useParseInternalPrefix**: this option remove prefix "_" for className, fileName etc. ex: _Role.ts will be Role.ts

**skipCollections**: specify which collections you want to generate from the schema. For example, we don't want to use Session

**deleteExistingFolder**: allow folder re-rewriting when generating. I think this is not the best option thought

**useClassDeclarationMerging**: allow us to declare class like below and then use them somewhere else. It is useful for us because we already had JS class but not the attributes available when typing etc.

```javascript
export interface AddressAttributes {
    id: string;
    objectId: string;
    createdAt: Date;
    updatedAt: Date;
    street: string;
    postalCode: string;
    city: string;
}

export interface Address extends AddressAttributes { }
export class Address extends AnotherClass or Parse.Object {
    constructor() {
        super('Address')
    }
}

Parse.Object.registerSubclass('Address', Address)

```

Feel free to propose other options to implement or re-think the ones above

Regards,

Anthony

**Comment by nicovitt** [comment]
Yes, the package is still maintained. I made some updates and created a new version. Are your suggestions still active? I can have a look at them. Would be best if you open a new ticket.
